From 7b82ce4a8457639d225449f3d345138c9f4f5ad9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Renaud=20M=C3=A9trich?=
 <1163635+rmetrich@users.noreply.github.com>
Date: Mon, 3 Sep 2018 05:42:39 +0200
Subject: [PATCH] journald: fixed assertion failure when system journal
 rotation fails (#9893)

(cherry picked from commit fd790d6f09b10a87b007b71403cb018f18ff91c9)
Resolves: #1619543
---
 src/journal/journald-server.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/journal/journald-server.c b/src/journal/journald-server.c
index 88d8f3e41d..88cf0b2d53 100644
--- a/src/journal/journald-server.c
+++ b/src/journal/journald-server.c
@@ -1180,7 +1180,8 @@ int server_flush_to_var(Server *s, bool require_flag_file) {
         }
 
 finish:
-        journal_file_post_change(s->system_journal);
+        if (s->system_journal)
+                journal_file_post_change(s->system_journal);
 
         journal_file_close(s->runtime_journal);
         s->runtime_journal = NULL;
